packit/copr_helper.py

```diff
@@ -134,6 +110,95 @@ def create_copr_project_if_not_exists(
                 additional_packages=additional_packages,
                 additional_repos=additional_repos,
             )
+            return
+
+        delete_after_days = -1 if preserve_project else 60
+
+        fields_to_change = self.get_fields_to_change(
+            copr_proj=copr_proj,
+            additional_repos=additional_repos,
+            chroots=chroots,
+            description=description,
+            instructions=instructions,
+            list_on_homepage=list_on_homepage,
+            delete_after_days=delete_after_days,
+        )
+
+        if fields_to_change:
+            logger.info(f"Updating copr project '{owner}/{project}'")
+            for field, (old, new) in fields_to_change.items():
+                logger.debug(f"{field}: {old} -> {new}")
+
+            try:
+                self.copr_client.project_proxy.edit(
+                    ownername=owner,
+                    projectname=project,
+                    chroots=chroots,
+                    description=description,
+                    instructions=instructions,
+                    unlisted_on_hp=not list_on_homepage,
+                    additional_repos=additional_repos,
+                    delete_after_days=delete_after_days,
+                )
+            except CoprRequestException as ex:
+                if "Only owners and admins may update their projects." in str(ex):
+                    logger.info(
+                        f"We have to be admins to be able to edit project settings. "
+                        f"Requesting the admin rights for the copr '{owner}/{project}' project."
+                    )
+                    copr_proj.request_permissions(
```

Comment:
Shouldn't this be configurable and be disabled by default?

As a packit CLI user I would be surprised and probably somewhat upset if this would happen without an explicit ask. I would rather see an error and directed to use some option to explicitly ask for this happening.

Reply:
> Shouldn't this be configurable and be disabled by default?

can be

> As a packit CLI user I would be surprised and probably somewhat upset if this would happen without an explicit ask. I would rather see an error and directed to use some option to explicitly ask for this happening.

How do you know? The thing is that you can't grant admin rights when the user doesn't ask for it. But yes, we can create a CLI/API option that will be forwarded.

Reply:
Let's say someone is working on one of the convert2rhel projects, which have custom Copr configuration. They want to trigger a build from a local checkout using `packit copr-build`. They have their personal Copr credentials configured, which don't have admin access to the configured Copr projects. They completely forget that `packit copr-build` will not build in their personal repositories, when they run the command they'll find themselves with this request happening. They did not ask for it, they just wanted to have a build.

Reply:
- I've implemented the CLI option.
- The request does not need to be approved.
- I consider this a really rear case but we should be fine now with the option.